### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by double-hope**

### CODE REVIEW

File: packages/frontend/components/primitives/search/component.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+const Search = ({ value, setValue, ...props }: SearchProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const { searchedProducts } = useTypedSelector((state) => state.product);
+  const { push } = useRouter();
+
+  const callback = async ({ target }) => {
+    setValue(target.value);
+    await dispatch(actionSearch(target.value.toString()));
```
Comment (dannleed) [inline packages/frontend/components/primitives/search/component.tsx:19]: maybe we can add debounce or throttle function here to send a search request with some interval but not on every onChange action, for example every 100ms or other value that will fit

File: packages/frontend/components/primitives/search/component.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  useEffect(() => {
+    if (searchedProducts.length) {
+      setIsOpen(true);
+    } else {
+      setIsOpen(false);
+    }
+  }, [searchedProducts]);
+
+  const redirectToItem = async (productId: string) => {
+    await dispatch(actionSearch(''));
```
Comment (dannleed) [inline packages/frontend/components/primitives/search/component.tsx:31]: if you need clear state i think sending empty request not the best idea
create clearSearch action or handle empty string in existing action

File: packages/mobile/src/screens/home/home.tsx [hunk 3]
```diff
@@ -79,12 +79,12 @@ const Home: FC = () => {
             </Text>
             <Flag />
           </View>
-          <SearchInput
+          {/* <SearchInput
             placeHolder={t('home.SEARCH_PLACEHOLDER')}
```
Comment (dannleed) [inline packages/mobile/src/screens/home/home.tsx:83]: why you do this?))

**Comment by double-hope** [comment]
image
image

**Comment by double-hope** [comment]
image

**Comment by double-hope** [comment]
image
image
screen with search button

**Review by dannleed — APPROVED** [review]

**Review by voonya — APPROVED** [review]

File: packages/frontend/components/header/component.tsx [hunk 4]
```diff
@@ -25,6 +25,10 @@ interface RequestOptions {
 }
 
 export const Header = () => {
+  const Search = dynamic(
+    () => import('@components/primitives/search/component'),
+  );
+
```
Comment (Nightlngale) [inline packages/frontend/components/header/component.tsx:31]: Use it outside of the component. Add input/icon as a suspense

**Review by Nightlngale — CHANGES_REQUESTED** [review]

**Review by Nightlngale — APPROVED** [review]

**Pull request merged**